src/m365/spo/commands/page/page-set.ts

```diff
@@ -343,7 +361,29 @@ class SpoPageSetCommand extends SpoCommand {
         await request.post(requestOptions);
       }
 
-      let requestOptions: any;
+      if (args.options.demoteFrom) {
+        const fileGetOptions: spoFileGetOptions = {
```

Comment:
It's not really applicable now, but let's put all this logic in an extra `if (args.options.demotefrom === 'NewsPage')` so we can extend it easily with future values.